source/jormungandr/jormungandr/interfaces/v1/Journeys.py

```diff
@@ -294,9 +294,24 @@ def wrapper(*args, **kwargs):
                 return objects
             for journey in objects[0]['journeys']:
                 args = dict(request.args)
-                ids = {o['stop_point']['stop_area']['id']
+                allowed_ids = {o['stop_point']['id']
                        for s in journey.get('sections', []) if 'from' in s
                        for o in (s['from'], s['to']) if 'stop_point' in o}
+
+                #only one first_section_mode is allowed if present in the request
+                #same for last_section_mode also
+                if next(iter(journey['sections'][1:]), {}).get('type', '').startswith('bss'):
+                    args['first_section_mode[]'] = 'bss'
+                else:
+                    args['first_section_mode[]'] = journey['sections'][0]['mode']
+
+                if next(iter(journey['sections'][-2::-1]), {}).get('type', '').startswith('bss'):
+                    args['last_section_mode[]'] = 'bss'
+                else:
+                    args['last_section_mode[]'] = journey['sections'][-1]['mode']
+
+                args['min_nb_transfers'] = journey['nb_transfers']
```

Comment:
It will not be enough for direct journey, as there is 0 transfers in direct pt journeys and non pt journeys. I don't have easy solution for that just now.